File: repos_estudiantes_ALVAREZ_primera_entrega/HUANCA_CÓNDORI_KEYLA_ESTRELLA/Practica1/POO/ej2/Bus.java
public class Bus {
    private int cantPasajeros;
    private int capacidadMaxima;
    private double totalRecaudado;
    private  double pasaje = 1.50;
    public Bus(int capacidadMaxima) {
        this.capacidadMaxima = capacidadMaxima;
        this.cantPasajeros = 0;
        this.totalRecaudado = 0;
    }
    public void subirPasajeros(int cantidad) {
        if (cantidad <= 0) {
            System.out.println("La cantidad debe ser mayor a 0");
            return;
        }
        int asientosDisponibles = capacidadMaxima - cantPasajeros;

        if (cantidad <= asientosDisponibles) {
            cantPasajeros += cantidad;
            System.out.println(cantidad + " pasajeros subieron al bus.");
        } else {
            cantPasajeros = capacidadMaxima;
            System.out.println("Solo pudieron subir " + asientosDisponibles + " pasajeros. El bus está lleno.");
        }
    }
    public void cobrarPasaje(int cantidadPasajeros) {
        if (cantidadPasajeros <= 0) {
            System.out.println("La cantidad debe ser mayor a 0");
            return;
        }

        double totalCobrado = cantidadPasajeros * pasaje;
        totalRecaudado += totalCobrado;
        System.out.println("Se cobró Bs. " + totalCobrado + " a " + cantidadPasajeros + " pasajeros.");
    }
    public void mostrarAsientosDisponibles() {
        int disponibles = capacidadMaxima - cantPasajeros;
        System.out.println("Asientos disponibles: " + disponibles + " de " + capacidadMaxima);
    }
    

}

File: repos_estudiantes_ALVAREZ_primera_entrega/HUANCA_CÓNDORI_KEYLA_ESTRELLA/Practica1/POO/ej2/bus.py
class Bus:
    def __init__(self, capacidadMaxima):
        self.cantPasajeros = 0
        self.capacidadMaxima = capacidadMaxima
        self.totalRecaudado = 0
        self.pasaje = 1.50

    def subirPasajeros(self, cantidad):
        if cantidad <= 0:
            print("La cantidad debe ser mayor a 0")
            return

        asientosDisponibles = self.capacidadMaxima - self.cantPasajeros

        if cantidad <= asientosDisponibles:
            self.cantPasajeros += cantidad
            print(f"{cantidad} pasajeros subieron al bus.")
        else:
            self.cantPasajeros = self.capacidadMaxima
            print(f"Solo pudieron subir {asientosDisponibles} pasajeros. El bus está lleno.")

    def cobrarPasaje(self, cantidadPasajeros):
        if cantidadPasajeros <= 0:
            print("La cantidad debe ser mayor a 0")
            return

        totalCobrado = cantidadPasajeros * self.pasaje
        self.totalRecaudado += totalCobrado
        print(f"Se cobró Bs. {totalCobrado} a {cantidadPasajeros} pasajeros.")

    def mostrarAsientosDisponibles(self):
        disponibles = self.capacidadMaxima - self.cantPasajeros
        print(f"Asientos disponibles: {disponibles} de {self.capacidadMaxima}")

    def mostrarInformacion(self):
        print("\ninformacion")
        print(f"Pasajeros actuales: {self.cantPasajeros}")
        print(f"Capacidad máxima: {self.capacidadMaxima}")
        print(f"Total recaudado: Bs. {self.totalRecaudado}")
        self.mostrarAsientosDisponibles()
miBus = Bus(50)
print("\nsubiendo pasajeros")
miBus.subirPasajeros(30)
miBus.mostrarAsientosDisponibles()
print("\n--- Cobrando pasajes ---")
miBus.cobrarPasaje(30)
miBus.mostrarInformacion()